crypto/kem/README.md

```diff
@@ -0,0 +1,159 @@
+                                                           ------------------->       
+                                                                                  // ENCAPSULATION
+                                                                                  // Set the received raw public key.
+                                                                                  EVP_PKEY *key;
+                                                                                  key = EVP_PKEY_kem_new_raw_public_key(NID_KYBER512,
+                                                                                                                        pub_key, pub_key_len);
+
+                                                                                  // Create the context.
+                                                                                  EVP_PKEY_CTX *encaps_ctx;
+                                                                                  encaps_ctx = EVP_PKEY_CTX_new(key, NULL);
+
+                                                                                  // Get the required buffer lengths and allocate the buffers.
+                                                                                  uint8_t *ct, *ss;      // ciphertext and shared secret,
+                                                                                  size_t ct_len, ss_len; // and their lengths.
+                                                                                  EVP_PKEY_encapsulate(encaps_ctx, NULL, &ct_len, NULL, &ss_len);
+                                                                                  ct = OPENSSL_malloc(ct_len);
+                                                                                  ss = OPENSSL_malloc(ss_len);
+
+                                                                                  // Encapsulate.
+                                                                                  EVP_PKEY_encapsulate(encaps_ctx, ct, &ct_len, ss, &ss_len);
+                                                                    ct                
+                                                           <-------------------
+// DECAPSULATION
+// Create the context.
+EVP_PKEY_CTX *decaps_ctx;
+decaps_ctx = EVP_PKEY_CTX_new(key, NULL);
+
+// Get the required buffer length and allocate the buffer.
+uint8_t *ss;   // shared secret,
+size_t ss_len; // and its length.
+
+EVP_PKEY_decapsulate(decaps_ctx, NULL, &ss_len, ct, ct_len);
+ss = OPENSSL_malloc(ss_len);
+
+// Decapsulate.
+EVP_PKEY_decapsulate(decaps_ctx, ss, &ss_len, ct, ct_len);
+
+// END                          Alice's shared secret is equal to Bob's shared secret if the protocol succeeded.
+
+
+```
+Notes about the code snipet above:
+* The functions used in the example can fail. For brevity, we omit the success/failure checks, but they should definitely be added to any code using the API.
```

Comment:
How bad is adding all the error checks? If I was a totally new developer could I copy/paste this code and get in trouble down the line?

Reply:
done